include/pinocchio/algorithm/model.hxx

```diff
@@ -221,6 +221,24 @@ namespace pinocchio
          model, geomModel);
   }
 
+  // Compute whether Joint child is a descendent of parent in a given model
+  // Joints are represented by their id in the model
+  template <typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  static bool hasAncestor(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                          JointIndex child, JointIndex parent)
+  {
+    // Any joints has universe as an acenstor
+    if (parent == 0) return true;
+    if (child == parent) return true;
+    JointIndex i = child;
+    while (i != 0)
+    {
+      i = model.parents[i];
+      if (i == parent) return true;
+    }
+    return false;
+  }
+
```

Comment:
Could you rather use `model.supports[child]` to check whether parent is supporting joint of `child`?

Reply:
It seems to me that the documentation of `model.supports` is incorrect. I would expect:
"supports[j] corresponds to the collection of all joints located on the path between joint *j*  and the root."
Is it correct ? If so, I will fix the doc in this PR.